lib/reservation_manager.rb

```diff
@@ -0,0 +1,111 @@
+require 'date'
+require_relative 'reservation_dates'
+require_relative 'reservation'
+
+module HotelBookings
+  class Reservation_Manager
+    MAX_ROOMS = 20
+    attr_reader :customer_name, :checkin, :checkout
+   
+    def initialize(customer_name:)
+      @customer_name = customer_name
+
+      @current_reservations = {1 => [], 2 => [], 3 => [], 4 => [], 5 => [], 6 => [], 7 => [], 8 => [], 9 => [], 10 => [], 11 => [], 12 => [], 13 => [], 14 => [], 15 => [], 16 => [], 17 => [], 18 => [], 19 => [], 20 => []}
+    end 
+
+    def rooms 
+      return @current_reservations
+    end 
+
+    def reservation_list(date)
+      unavailable_rooms = []
+      key = 1
+      MAX_ROOMS.times do
+        if @current_reservations[key].include?(date) == true 
+          unavailable_rooms.push(key)
+        end 
+        key += 1
+      end
+      unavailable_rooms = unavailable_rooms.uniq
+      return unavailable_rooms
+    end
+  
+    def available_rooms_list(date)
+      available_rooms = []
+      key = 1
+      MAX_ROOMS.times do 
```

Comment:
Using a `.times` loop feels awkward, and relying on a variable called `key` doesn't quite evoke the meaning of what `key` is. It might be interesting to refactor this to use a `.select`, since `.select` will create an array of values that are true in certain conditions (like if `@current_reservations` at that room excludes the date)